lib/authentication-context.js

```diff
@@ -248,7 +248,8 @@ AuthenticationContext.prototype.acquireToken = function(resource, userId, client
  * @param {string}   clientId                            The OAuth client id of the calling application.
  * @param {AcquireTokenCallback}   callback              The callback function.
  */
-AuthenticationContext.prototype.acquireTokenWithUsernamePassword = function(resource, username, password, clientId, callback)  {
+AuthenticationContext.prototype.acquireTokenWithUsernamePassword = function(resource, username, password, clientId, clientSecret, callback)  {
```

Comment:
@RandalliLama I thought I'd go back and confirm I needed to include the client secret to successfully authenticate. I'm trying to authenticate against the https://graph.microsoft.com endpoint with this package. If I just use the `acquireTokenWithClientCredentials` function I get back a token but it doesn't include anything about the user. Thus I get an `unauthorized` response when I try to make API calls with the token I got back. However, when I try authenticating with the original `acuireTokenWithUsernameAndPassword` function I get an error from the API that says I need to include the client secret. As soon as I included the client secret when I acquire the token I'm able to successfully authenticate and use the API. 

I left the original tests for the method and made changes so that if I add in the `clientSecret` then it will use it. Otherwise it falls back to the original implementation. I'm nervous about writing a separate method since it requires I need to write 3 new methods, not just one. Also, I don't know enough about the federated user workflow. Should I add the client secret to that workflow as well?

Thanks for you help.